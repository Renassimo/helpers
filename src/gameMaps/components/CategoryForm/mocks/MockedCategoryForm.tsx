const MockedCategoryForm = jest.fn(({ values, errors, isReady }) => (
  <div>
    MockedCategoryForm - {values?.title} - {values?.description} -{' '}
    {values?.color} - {values?.itemsAmount} - {errors?.title} -{' '}
    {errors?.description} - {errors?.color} - {errors?.itemsAmount} -{' '}
    {isReady && 'isReady'}
  </div>
));

export default MockedCategoryForm;
