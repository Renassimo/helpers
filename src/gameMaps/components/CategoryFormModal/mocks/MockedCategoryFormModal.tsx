const MockedCategoryFormModal = jest.fn(({ isModalOpen, gameId }) => (
  <div>
    MockedCategoryFormModal {isModalOpen ? 'opened' : 'closed'} - {gameId}
  </div>
));

export default MockedCategoryFormModal;
