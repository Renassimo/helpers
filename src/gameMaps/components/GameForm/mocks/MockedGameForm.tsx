const MockedGameForm = jest.fn(({ values, errors }) => (
  <div>
    MockedGameForm - {values?.title} - {values?.description} -{' '}
    {values?.backgroundColor} - {values?.mapImageUrl} - {errors?.title} -{' '}
    {errors?.description} - {errors?.backgroundColor}
  </div>
));

export default MockedGameForm;
