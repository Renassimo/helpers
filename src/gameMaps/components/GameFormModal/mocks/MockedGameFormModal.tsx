const MockedGameFormModal = jest.fn(({ isModalOpen }) => (
  <div>MockedGameFormModal {isModalOpen ? 'opened' : 'closed'}</div>
));

export default MockedGameFormModal;
