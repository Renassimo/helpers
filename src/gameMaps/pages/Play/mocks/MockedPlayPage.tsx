const MockedPlayPage = jest.fn(() => <div>MockedPlayPage</div>);

export default MockedPlayPage;
