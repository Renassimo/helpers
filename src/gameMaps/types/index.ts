export * from './gameTypes';
export * from './playTypes';
