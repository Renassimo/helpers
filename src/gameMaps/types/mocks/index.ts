export * from './gameMocks';
export * from './playMocks';
export * from './categoryMocks';
export * from './itemMocks';
