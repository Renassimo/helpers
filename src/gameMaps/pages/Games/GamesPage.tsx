import { GamesPageProps } from '@/gameMaps/types';
import { PageInfo } from '@/auth/types';

import { useErrorAlert } from '@/common/hooks/alerts';

import GameMapsTemplate from '@/gameMaps/templates/GameMapsTemplate';

import PagesList from '@/common/components/PagesList';

const GamesPage = ({ user, pages, data, error }: GamesPageProps) => {
  useErrorAlert(error);

  const gamePages: PageInfo[] = [
    ...(data?.map((game) => ({
      title: game.attributes.title,
      path: `/gameMaps/games/${game.id}`,
    })) ?? []),
    {
      title: '+ Add New Game',
      path: '/gameMaps/games/new',
      onClick: () => console.log('New!'),
    },
  ];

  return (
    <GameMapsTemplate
      title="Game Maps"
      user={user}
      pages={pages}
      description="Game Maps"
    >
      {data && <PagesList pages={gamePages} />}
    </GameMapsTemplate>
  );
};

export default GamesPage;
