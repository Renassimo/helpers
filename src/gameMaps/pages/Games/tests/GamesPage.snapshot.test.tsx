import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import { PageInfo } from '@/common/types/auth';

import PageTemplate from '@/common/templates/PageTemplate';
import PagesList from '@/common/components/PagesList';

import GamesPage from '../GamesPage';

import MockedPageTemplate from '@/common/templates/PageTemplate/mocks';
import MockedPagesList from '@/common/components/PagesList/mocks';
import { mockedUser } from '@/common/types/auth/mocks';
import { mockedGames } from '@/gameMaps/types/mocks';

jest.mock('@/common/templates/PageTemplate');
jest.mock('@/common/components/PagesList');

describe('GamesPage snapshot', () => {
  const mockedPages: PageInfo[] = [{ title: 'Page 1', path: '/page1' }];

  beforeAll(() => {
    (PageTemplate as unknown as jest.Mock).mockImplementation(
      MockedPageTemplate
    );
    (PagesList as unknown as jest.Mock).mockImplementation(MockedPagesList);
  });
  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <GamesPage
        data={mockedGames}
        user={mockedUser}
        error={null}
        pages={mockedPages}
      />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
