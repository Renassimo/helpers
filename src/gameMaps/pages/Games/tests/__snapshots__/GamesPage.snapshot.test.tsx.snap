// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesPage snapshot renders successfully 1`] = `
<body>
  <div>
    <div>
      Mocked Page Template (
      Game Maps
      ):
      <span>
        Jesse Pinkman
         - 
        Page 1
      </span>
      <h1
        class="MuiTypography-root MuiTypography-h5 css-1uxccsd-MuiTypography-root"
      >
        Games Page
      </h1>
      <div>
        <a
          href="/gameMaps/games/gm1"
        >
          Game 1
        </a>
        <a
          href="/gameMaps/games/gm2"
        >
          Game 2
        </a>
        <a
          href="/gameMaps/games/new"
        >
          + Add New Game
        </a>
      </div>
    </div>
  </div>
</body>
`;
